src/TorchSharp/NN/Sequential.cs

```diff
@@ -176,6 +181,41 @@ public override nn.Module apply(Action<nn.Module> fn)
                 }
             }
 
+            /// <summary>
+            /// Module indexer.
+            /// </summary>
+            [IndexerName("SequentialItems")]
+            public Sequential this[(int? start, int? end) index] {
+                get {
+                    var result = new Sequential(Array.Empty<torch.nn.Module<Tensor, Tensor>>());
+                    var start = index.start.HasValue ? index.start.Value : 0;
+                    var end = index.end.HasValue ? index.end.Value : _modules.Count;
```

Comment:
>index.end.Value [](http://example.com/codeflow?start=51&length=15)

You may make this `Math.Min(index.end.Value, _modules.Count)` and then you can remove the condition `i < _modules.Count ` from the loop.